src/app/signers/SignersMap.tsx

```diff
@@ -0,0 +1,398 @@
+    md: { center: [37, -100], zoom: 3 },
+    xl: { center: [33, -95], zoom: 3 },
+  }) as CenterAndZoom;
+  const africaCenterAndZoom = useBreakpointValue({
+    // TODO: tune
+    base: { center: [9, 18], zoom: 3 },
+    md: { center: [9, 18], zoom: 3 },
+    xl: { center: [9, 18], zoom: 3 },
+  }) as CenterAndZoom;
+  const europeCenterAndZoom = useBreakpointValue({
+    base: { center: [53, 10], zoom: 3 },
+    md: { center: [48, 0], zoom: 4 },
+    xl: { center: [46, 12], zoom: 4 },
+  }) as CenterAndZoom;
+  const australiaCenterAndZoom = useBreakpointValue({
+    // TODO: tune
+    base: { center: [-25, 133], zoom: 4 },
+    md: { center: [-25, 133], zoom: 4 },
+    xl: { center: [-25, 133], zoom: 4 },
+  }) as CenterAndZoom;
+  const southAmericaCenterAndZoom = useBreakpointValue({
+    // TODO: tune
+    base: { center: [-20, -61], zoom: 3 },
+    md: { center: [-20, -61], zoom: 3 },
+    xl: { center: [-20, -61], zoom: 3 },
+  }) as CenterAndZoom;
+  const asiaCenterAndZoom = useBreakpointValue({
+    base: { center: [35, 100], zoom: 2 },
+    md: { center: [20, 150], zoom: 3 },
+    xl: { center: [17, 115], zoom: 3 },
+  }) as CenterAndZoom;
+
+  const getContinetCenterAndZoom = useCallback(
+    (continent: Continent | null): CenterAndZoom => {
+      switch (continent) {
+        case Continent.NorthAmerica:
+          return northAmericaCenterAndZoom;
+        case Continent.Africa:
+          return africaCenterAndZoom;
+        case Continent.Europe:
+          return europeCenterAndZoom;
+        case Continent.Australia:
+          return australiaCenterAndZoom;
+        case Continent.SouthAmerica:
+          return southAmericaCenterAndZoom;
+        case Continent.Asia:
+          return asiaCenterAndZoom;
+        default:
+          return { center: INITIAL_VIEW.center, zoom: INITIAL_VIEW.zoom } as CenterAndZoom;
+      }
+    },
+    [
+      northAmericaCenterAndZoom,
+      africaCenterAndZoom,
+      europeCenterAndZoom,
+      australiaCenterAndZoom,
+      southAmericaCenterAndZoom,
+      asiaCenterAndZoom,
+    ]
+  );
+
+  useEffect(() => {
+    if (prevActiveContinent.current === activeContinent) return;
+    prevActiveContinent.current = activeContinent;
+    const centerAndZoom = getContinetCenterAndZoom(activeContinent);
+    const { center, zoom } = centerAndZoom;
+    map.setView(center, zoom);
+    if (activeContinent === null) {
+      setActiveMarkerLatLng(undefined);
+    }
+  }, [map, activeContinent, getContinetCenterAndZoom, setActiveMarkerLatLng]);
+
+  return null;
+};
+
+interface MarkerLatLng {
+  lat: number;
+  lng: number;
+}
+
+const MarkerClickHandler = ({ markerLatLng }: { markerLatLng: MarkerLatLng | undefined }) => {
+  const map = useMap();
+
+  const prevMarkerLatLng = useRef<MarkerLatLng | undefined>(markerLatLng);
+  const newMarkerClicked =
+    (prevMarkerLatLng.current === undefined && markerLatLng !== undefined) ||
+    (prevMarkerLatLng.current &&
```

Comment:
would you mind putting in a comment for this part or maybe define the condition in a separate variable with a descriptive name to make it easier to understand?

Reply:
done